platform-modules/batch-models/src/main/scala/org/ekstep/analytics/updater/UpdatePortalMetrics.scala

```diff
@@ -0,0 +1,88 @@
+  * @author Manjunath Davanam <manjunathd@ilimi.in>
+  */
+
+import com.datastax.spark.connector._
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.ekstep.analytics.adapter.ContentAdapter
+import org.ekstep.analytics.framework._
+import org.ekstep.analytics.framework.dispatcher.AzureDispatcher
+import org.ekstep.analytics.framework.util.{CommonUtil, JSONUtils}
+import org.ekstep.analytics.util.{Constants, WorkFlowUsageSummaryFact}
+
+import scala.util.Try
+
+
+case class WorkflowSummaryEvents(deviceId: String, mode: String, dType: String, totalSession: Long, totalTs: Double) extends AlgoInput with Input
+
+case class Metrics(noOfUniqueDevices: Long, totalContentPlaySessions: Double, totalTimeSpent: Double, totalContentPublished: Long) extends AlgoOutput with Output
+
+case class PortalMetrics(eid: String, ets: Long, syncts: Long, metrics_summary: Option[Metrics]) extends AlgoOutput with Output
+
+object UpdatePortalMetrics extends IBatchModelTemplate[DerivedEvent, WorkflowSummaryEvents, Metrics, PortalMetrics] with Serializable {
+
+  val className = "org.ekstep.analytics.updater.UpdatePortalMetrics"
+
+  private val EVENT_ID: String = "ME_PORTAL_CUMULATIVE_METRICS"
+
+  override def name: String = "UpdatePortalMetrics"
+
+  /**
+    * preProcess which will fetch the `WorkFlowUsageSummaryFact` Event data from the Cassandra Database.
+    *
+    * @param data   - RDD Event Data(Empty RDD event)
+    * @param config - Configurations to run preProcess
+    * @param sc     - SparkContext
+    * @return - workflowSummaryEvents
+    */
+  override def preProcess(data: RDD[DerivedEvent], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[WorkflowSummaryEvents] = {
+    sc.cassandraTable[WorkFlowUsageSummaryFact](Constants.PLATFORM_KEY_SPACE_NAME, Constants.WORKFLOW_USAGE_SUMMARY_FACT).filter { x => x.d_period == 0 }.map(event => {
+      WorkflowSummaryEvents(event.d_device_id, event.d_mode, event.d_type, event.m_total_sessions, event.m_total_ts)
+    })
+  }
+
+  /**
+    *
+    * @param data   - RDD Workflow summary event data
+    * @param config - Configurations to algorithm
+    * @param sc     - Spark context
+    * @return - DashBoardSummary ->(uniqueDevices, totalContentPlaySession, totalTimeSpent,)
+    */
+  override def algorithm(data: RDD[WorkflowSummaryEvents], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[Metrics] = {
+    object _constant extends Enumeration {
+      val APP = "app"
+      val PLAY = "play"
+      val CONTENT = "content"
+      val SESSION = "session"
+      val ALL = "all"
+    }
+    val uniqueDevicesCount = data.filter(x => x.deviceId != _constant.ALL).map(_.deviceId).distinct().count()
+    val totalContentPlaySession = data.filter(x => x.mode.equals(_constant.PLAY) && x.dType.equals(_constant.CONTENT)).map(_.totalSession).sum()
+    val totalTimeSpent = data.filter(x => x.dType.equals(_constant.APP) || x.dType.equals(_constant.SESSION)).map(_.totalTs).sum()
```

Comment:
@manjudr  The totalTimeSpent should be in hours. Please convert the result to hours.

Reply:
Converted.